linux/hci/hci.go

```diff
@@ -330,6 +327,25 @@ func (h *HCI) checkOpCodeFree(opCode int) error {
 	return nil
 }
 
+func (h *HCI) getTxBuf() ([]byte, error) {
+	// get buffer w/timeout
+	var b []byte
+	select {
+	case <-h.done:
+		return nil, fmt.Errorf("hci closed")
+	case b = <-h.chCmdBufs:
+		//ok
+	case <-time.After(chCmdBufTimeout):
+		err := fmt.Errorf("chCmdBufs get timeout")
+		h.dispatchError(err)
+		return nil, err
+	}
+
+	return b, nil
+}
+
+var ocl = &opCodeLocker{}
```

Comment:
might be better if this was in the HCI struct

Reply:
Ok, I can move it there